mobile/src/src/components/Menu/Profile/EditComponent.jsx

```diff
@@ -0,0 +1,691 @@
+        pronouns: zeroPronoun == 0 ? 0 : pronoun ? pronoun : user.pronouns,
+        birthday: birth ? birth : user.birthday,
+        region: zeroReg == 0 ? 0 : reg ? reg : user.region,
+        enrolled_program:
+          zeroPro == 0 ? 0 : program ? program : user.enrolled_program,
+        current_term: zeroTerm == 0 ? 0 : term ? term : user.current_term,
+      },
+      (response) => {
+        navigation.navigate("Profile");
+      },
+      (error) => {
+        console.log(error);
+        console.log(error.response);
+      }
+    );
+  };
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== "web") {
+        const status = await ImagePicker.requestCameraRollPermissionsAsync();
+        if (status !== "granted") {
+          alert("You need camera roll permissions to change your pictures");
+        }
+      }
+    })();
+  }, []);
+
+  // used 2 pick image functions or else the photo library appears by itself
+  const pickImage1 = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+
+    console.log(result);
+    console.log("here1");
+    console.log(result.uri);
+
+    if (!result.cancelled) {
+      setPfp(result.uri);
+      console.log("here");
+    }
+  };
+
+  const pickImage2 = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+
+    console.log(result);
+    console.log("here1");
+    console.log(result.uri);
+
+    if (!result.cancelled) {
+      setbgPic(result.uri);
+      console.log("here");
+    }
+  };
+
+  function getOptions(type) {
+    var temp = [];
+    var tempChoice =
+      type == "Pronoun"
+        ? Choices.pronouns
+        : type == "Region"
+        ? Choices.regions
+        : type == "Program"
+        ? Choices.programs
+        : type == "Current Term"
+        ? Choices.terms
+        : "";
+
+    if (tempChoice.length) {
+      for (let i = 0; i < tempChoice.length; i++) {
+        temp.push(
+          <Picker.Item
+            key={i}
+            label={tempChoice[i]}
+            value={i}
+            style={styles.modalOptions}
+          />
+        );
+      }
+    }
+
+    return temp;
+  }
+
+  const onPickerChange = (itemValue, itemIndex, type) => {
+    var zero = 999;
+    if (itemValue == 0) {
+      zero = 0;
+    }
+    if (type == "Pronoun") {
+      setZeroPronoun(zero);
+      setPronoun(itemValue);
+    } else if (type == "Region") {
+      setZeroReg(zero);
+      setRegion(itemValue);
+    } else if (type == "Program") {
+      setZeroPro(zero);
+      setProgram(itemValue);
+    } else if (type == "Current Term") {
+      setZeroTerm(zero);
+      setTerm(itemValue);
+    }
+  };
+
+  function getSpecificModal(type) {
```

Comment:
@shashankanandkannan and I used the Picker here, its also in a modal bc on ios its scrolling but on android it should be like a dropdown